File: application/backend/app/models/category.py
from sqlalchemy import Column, String
from sqlalchemy.orm import relationship
from .base import BaseModel

class Category(BaseModel):

    __tablename__ = "categories"
    
    name = Column(String(50), unique=True, nullable=False, index=True)
    description = Column(String(200), nullable=True)
    color = Column(String(7), default="#007bff")
    
    # Relations
    tasks = relationship("Task", back_populates="category", cascade="all, delete-orphan")
    
    def __repr__(self):
        return f"<Category(name={self.name})>"

File: application/frontend/src/types/category.ts
export interface Category {
  id: number
  name: string
  description?: string
  color?: string
  task_count?: number
  created_at: string
  updated_at: string
}

export interface CreateCategoryRequest {
  name: string
  description?: string
  color?: string
}

export interface UpdateCategoryRequest {
  name?: string
  description?: string
  color?: string
}
